tuf/data/root.go

```diff
@@ -11,6 +11,7 @@ type SignedRoot struct {
 	Signatures []Signature
 	Signed     Root
 	Dirty      bool
+	Serializer Serializer
```

Comment:
This was basically my attempt at getting more coverage by doing dependency injection.  This may be just overkill, since we have to propagate JSON errors everywhere.  I'm happy to take it out, if we are ok with merging with slightly lower code coverage (because I'm adding lines that propagate JSON marshalling errors).

The two arguments I have in favor are:
- if we switch canonical JSON libraries, or if go adds one, that has a different API (specifying a canonical boolean for instance) then we can gloss over it with the serializer
- we can possibly at a later point, if we wanted to adopt this model, randomly changing the serializer in different pieces of metadata and seeing what will break and what won't

But again, happy to remove it.  It adds extra complexity for not a ton of gain.